ui/app/src/components/GitAuthForm/GitAuthForm.tsx

```diff
@@ -0,0 +1,259 @@
+            onKeyPress={onKeyPress}
+            onChange={handleInputChange}
+            InputProps={{
+              endAdornment: (
+                <InputAdornment position="end">
+                  <IconButton edge="end" aria-label="toggle password visibility" onClick={handleClickShowPassword}>
+                    {showPassword ? <VisibilityOff /> : <Visibility />}
+                  </IconButton>
+                </InputAdornment>
+              )
+            }}
+          />
+        )}
+        {type === 'key' && (
+          <TextField
+            id="repoKey"
+            name="repoKey"
+            label={formatMessage(messages.privateKey)}
+            required
+            fullWidth
+            multiline
+            className={classes.margin}
+            error={inputs.submitted && !inputs.repoKey}
+            helperText={renderHelperText(
+              formatMessage(messages.privateKey),
+              inputs.repoKey,
+              '',
+              true,
+              inputs.submitted
+            )}
+            onKeyPress={onKeyPress}
+            onChange={handleInputChange}
+            value={inputs.repoKey}
+          />
+        )}
+      </div>
+    );
+  }
+
+  return (
+    <RadioGroup
+      aria-label="repoAuthentication"
+      name="repoAuthentication"
+      value={inputs.repoAuthentication}
+      onChange={handleInputChange}
+      onKeyPress={onKeyPress}
+    >
+      <FormControlLabel
+        value="none"
+        control={<Radio onChange={() => viewAuth('none')} />}
+        label={
+          <FormattedMessage
+            id="gitForm.noAuthenticationRequired"
+            defaultMessage="Authentication not required (public URL)"
+          />
+        }
+      />
+      <FormControlLabel
+        value="basic"
+        control={<Radio onChange={() => viewAuth('basic')} />}
+        label={<FormattedMessage id="gitForm.usernameAndPassword" defaultMessage="Username & Password" />}
+      />
+      <Collapse in={inputs.expanded.basic} timeout={300} unmountOnExit>
+        {inputs.expanded.basic && renderAuth(inputs.repoAuthentication)}
+      </Collapse>
+      <FormControlLabel
+        value="token"
+        control={<Radio onChange={() => viewAuth('token')} />}
+        label={<FormattedMessage id="gitForm.token" defaultMessage="Token" />}
+      />
+      <Collapse in={inputs.expanded.token} timeout={300} unmountOnExit>
+        {inputs.expanded.token && renderAuth(inputs.repoAuthentication)}
+      </Collapse>
+      <FormControlLabel
+        value="key"
+        control={<Radio onChange={() => viewAuth('key')} />}
+        label={<FormattedMessage id="gitForm.privateKey" defaultMessage="Private Key" />}
+      />
+      <Collapse in={inputs.expanded.key} timeout={300} unmountOnExit>
+        {inputs.expanded.key && renderAuth(inputs.repoAuthentication)}
+      </Collapse>
+    </RadioGroup>
+  );
+}
```

Comment:
Please test create site thoroughly

Reply:
Tested with different types of validations (none, basic, key)